### ISSUE TITLE
Expo app crashes immediately on project startup

### ISSUE THREAD

**Issue by Spherrrical** [expo/expo#18860]
### Summary

Hello, I'm attempting to publish my app on Expo, however when I click the app name, it crashes immediately with no error. About a few days ago it was working fine with iOS 15, but I recently updated to iOS 16 Public Beta 4 and it seems to be crashing. I've tried running the app on a different device running iOS 15, but no luck as it keeps crashing.

### What platform(s) does this occur on?

iOS

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: Windows 10 10.0.19044
    Binaries:
      Node: 16.17.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 8.3.0 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      expo: ~46.0.7 => 46.0.7
      react: 18.0.0 => 18.0.0
      react-dom: 18.0.0 => 18.0.0
      react-native: 0.69.5 => 0.69.5
      react-native-web: ~0.18.7 => 0.18.7
    Expo Workflow: managed

### Minimal reproducible example

`Step 1: https://github.com/Spherrrical/CricketCounterV2`
`Step 2: npm install`
`Step 3: expo start/npm start`
`Step 4: Publish a branch with a project on Expo.dev`
`Step 5: Open the branch`

**Comment by BenEdgeContra** [comment]
I have the exact same issue. 
<img width="486" alt="image" src="https://user-images.githubusercontent.com/38164008/187085315-6f3db8fc-b9a8-443f-b573-8cec43a19577.png">

**Comment by Spherrrical** [comment]
Any updates on this anyone?

**Comment by douglowder** [comment]
Does the problem still occur with the released iOS 16?

**Comment by Spherrrical** [comment]
> Does the problem still occur with the released iOS 16?

Yes, but I found a workaround by creating a new app and copying all the code. Not sure why that works but it does.

**Comment by alex-fournier** [comment]
After spending days investigating, I opened a more specific issue: https://github.com/expo/expo/issues/19517

**Comment by douglowder** [comment]
Closing and will track this in the new more specific issue
